File: Api/Transfer/System.cs
using System;
using System.Collections.Immutable;

namespace Kafe.Api.Transfer;

public record SystemDetailDto(
    string Name,
    ImmutableArray<string> BaseUrls,
    string Version,
    string Commit,
    DateTimeOffset CommitDate,
    DateTimeOffset RunningSince
);

public record VideoConversionStatsDto(
    int TotalVideoShardCount,
    int CorruptedVideoShardCount,
    int PendingVideoConversionCount,
    int FailedVideoConversionCount
);

public record VideoConversionRetryDto(
    ImmutableArray<Hrib>? Ids = null,
    bool ShouldRetryOriginalAnalysis = false,
    bool ShouldRetryConversion = true
);

File: Web/src/data/System.ts
import { components } from '../schemas/api';
import { AbstractType } from './AbstractType';

export class System extends AbstractType {
    public name!: string;
    public version!: string;
    public commit!: string;
    public commitDate!: string;
    public runningSince!: string;

    public constructor(struct: components['schemas']['EntityPermissionsDetailDto']) {
        super();
        Object.assign(this, struct);
    }
}
